test/system/src/test/java/io/pravega/test/system/MultiReaderTxnWriterWithFailoverTest.java

```diff
@@ -387,12 +388,32 @@ private void performFailoverTest() {
                                                         final AtomicBoolean stopWriteFlag, final AtomicLong eventWriteCount) {
         return CompletableFuture.runAsync(() -> {
             while (!stopWriteFlag.get()) {
-                Transaction<Long> transaction = null;
+                Transaction<Long> txnDebugReference = null;
+
                 try {
-                    transaction = retry
+                    Transaction<Long> transaction = retry
                             .retryingOn(MultiReaderTxnWriterWithFailoverTest.TxnCreationFailedException.class)
                             .throwingOn(RuntimeException.class)
                             .run(() -> createTransaction(writer, stopWriteFlag));
+                    txnDebugReference = transaction;
+                    AtomicBoolean txnIsDone = new AtomicBoolean(false);
+
+                    // Sets a recurrent delayed task to ping the txn. It exits when the
+                    // txn completes and no longer needs pinging
+                    FutureHelpers.loop(txnIsDone::get, () -> {
+                        return FutureHelpers.delayedTask(() -> {
+                            if (transaction.checkStatus() == Transaction.Status.OPEN) {
+                                FutureHelpers.runOrFail(() -> {
+                                    transaction.ping(5000);
+                                    return null;
+                                }, new CompletableFuture<Void>());
+                            } else {
+                                txnIsDone.set(true);
+                            }
+
+                            return null;
+                        }, Duration.ofMillis(2000), executorService);
+                    }, executorService);
```

Comment:
Minor : we need to change line 421 from `transaction` to `transaction.getTxnId()`
https://github.com/pravega/pravega/pull/1766/files#diff-fd4ef3523065620e53c0447293088fd0R421